lib/agent/actions/wipe/windows.js

```diff
@@ -2,8 +2,55 @@ var os   = require('os'),
     join = require('path').join,
     exec = require('child_process').exec;
 
+var regispath = {
+  outlook_version: join('HKEY_CLASSES_ROOT', 'Outlook.Application', 'CurVEr'),
+  profileRegistry: join('HKEY_CURRENT_USER', 'Software', 'Microsoft'),
+  firefox:         join('HKEY_CLASSES_ROOT', 'FirefoxHTML', 'DefaultIcon'),
+  thunderbird:     join('HKEY_CLASSES_ROOT', 'ThunderbirdEML', 'DefaultIcon')
+};
+
+var registry = {
+  query:         function(out, cb) { exec('reg query ' + '"' + out + '"', cb) },
+  add:           function(out, cb) { exec('reg add '+ '"' + out + '"', cb) },
+  delete:        function(out, cb) { exec('reg delete '+ '"' + out + '"' +' /f ', cb) },
+  killtask:      function(out, cb) { exec('taskkill /F ' + out , cb) },
+  createProfile: function(out, cb) { exec('"' + out + '"' + ' -CreateProfile default', cb) }
+};
+
+var getOutlookVersion = function(cb) {
+  registry.query(regispath.outlook_version, function(err, stdout) {
+    if (err) return cb(err);
+    cb(err, stdout.split("\n")[2].split(".").pop());
+  });
+}
+
+var getProfileRegistry = function(cb) {
+  var profileReg;
+  this.getOutlookVersion(function(err, out){
+    if (err) return cb(err);
+    if (out.length == 1) {
+      profileReg = join(regispath.profileRegistry, 'Windows Messaging Subsystem', 'Profiles');
+    } else {
+      if (parseInt(out) >= 15) {
+        profileReg = join(regispath.profileRegistry, 'Office', out + '.0', 'Outlook', 'Profiles');
+      } else {
+        profileReg = join(regispath.profileRegistry, 'Windows NT', 'CurrentVersion', 'Windows Messaging Subsystem', 'Profiles');
+      }
+    }
+    cb(err, profileReg);
+  });
+}
+
+var getSoftwareDir = function(softwareName, cb) {
+  registry.query(regispath[softwareName], function(err, stdout) {
+    if (err) return cb(err);
+    cb(err, stdout.split("    ")[3].split(",")[0]);
```

Comment:
can this be achieved with a regexp? good thing about regex'es is that you can describe them better